raven/processes/wps_shape_area.py

```diff
@@ -0,0 +1,121 @@
+import logging
+
+import re
+import fiona
+from shapely.geometry import shape
+from pywps import LiteralInput, ComplexInput
+from pywps import LiteralOutput
+from pywps import Process
+
+LOGGER = logging.getLogger("PYWPS")
+
+
+class ShapeArea(Process):
+    """Given a file containing vector data, provide general information and spatial characteristics"""
+
+    def __init__(self):
+        inputs = [
+            LiteralInput('expand features', 'Examine all features in shapefile', data_type='boolean'),
+            ComplexInput('shape', 'Vector Shape',
+                         abstract='An URL pointing to either an ESRI Shapefile, GML, GeoJSON, or any other file in a'
+                                  ' standard vector format. The ESRI Shapefile must be zipped and contain the .shp,'
+                                  ' .shx, and .dbf. The shape CRS definition should also match the DEM CRS.',
+                         min_occurs=1,
+                         default=None, supported_formats=[])]
+        # supported_formats=[
+        #            # shp
+        #            {mimeType: 'application/zip',
+        #            encoding: '???',
+        #            schema: None},
+        #
+        #            # gml
+        #            {mimeType: 'text/xml',
+        #            encoding:'utf-8',
+        #            schema:'http://schemas.opengis.net/gml/3.2.1/gml.xsd'},
+        #
+        #            # json
+        #            {mimeType: 'text/plain',
+        #            encoding: 'iso-8859-2',
+        #            schema: None
+        #            },
+        #
+        #            # kml
+        #            {mimeType: 'text/xml',
+        #            encoding: 'windows-1250',
+        #            schema: 'http://schemas.opengis.net/kml/2.2.0/ogckml22.xsd'}
+        #            ])
+
+        outputs = [
+            LiteralOutput('area', 'Area Calculations', data_type='float', abstract='Area of shape in sq. km.'),
+            LiteralOutput('centroids', 'Centroid Locations', data_type='float',
+                          abstract="Geographic locations of feature centroids"),
+            LiteralOutput('schemas', 'Feature schemas',
+                          abstract='Geographic representations and descriptions of shape features')
```

Comment:
Not sure I understand what that means.

Reply:
The direct schema is the representation of fields and the data types they use (e.g. `{'geometry': Polygon, 'properties': {'field1': Int, 'field2': string, etc.} }`. This is more or less the internal structure of the shape/vector.